cnd/src/network/comit_ln.rs

```diff
@@ -306,18 +306,47 @@ impl NetworkBehaviourEventProcess<announce::behaviour::BehaviourOutEvent> for Co
     fn inject_event(&mut self, event: announce::behaviour::BehaviourOutEvent) {
         match event {
             announce::behaviour::BehaviourOutEvent::ReceivedAnnouncement { peer, mut io } => {
+                if let Some(local_swap_id) =
+                    self.swaps_waiting_for_announcement.get(&io.swap_digest)
+                {
+                    let create_swap_params = self.swaps.get(&local_swap_id).unwrap();
```

Comment:
Since you are touching this code can we clean up these `unwraps` while we are at it?  Either with and explicit check on the return value and a trace for error OR use an expect with a message saying why this should never be hit.

Reply:
I know that this is possible, but I would keep the if-let construct here. If we ever change the code we might overlook the `unwrap`. 
I assume that the if-let "makes you think" (because there is an if without an else close), but I don't see this as a big problem.